AwsCryptographicMaterialProviders/dafny/AwsCryptographyKeyStoreAdmin/src/DescribeMutation.dfy

```diff
@@ -84,6 +84,22 @@ module {:options "/functionSyntax:4" } DescribeMutation {
         + " Mutation Commitment UUID: " + Commitment.UUID + ";"
         + " Mutation Index UUID: " + Index.UUID + ";"
       ));
+    :- Need(
+      StateStrucs.ValidCommitment?(Commitment),
+      Types.MutationInvalidException(
+        message := "Mutation Commitment read from Storage is invalid or corrupted."
+        + " Recommend auditing the Branch Key's items for tampering."
+        + " Recommend auditing access to the storage."
+        + "\nBranch Key ID: " + input.Identifier + ";"
+        + " Mutation Commitment UUID: " + Commitment.UUID));
+    :- Need(
+      StateStrucs.ValidIndex?(Index),
+      Types.MutationInvalidException(
+        message := "Mutation Index read from Storage is invalid or corrupted."
+        + " Recommend auditing the Branch Key's items for tampering."
+        + " Recommend auditing access to the storage."
+        + "\nBranch Key ID: " + input.Identifier + ";"
+        + " Mutation Index UUID: " + Index.UUID));
```

Comment:
I don't know what these messages mean. Specifically the "Recommend" sentences. 
If there's a TODO to make these better can you note it here?

Reply:
There is no TODO for this, though there could be.
However, we do not have ANY documentation on what to do if a Branch Key has been tampered with,
much like we have no documentation on what to do if the auth tag check on an AES-GCM decrypt fails,
or if the asymmetric signature evaluation from a digital signing suite fails.

In general,
we throw an error,
and leave it to the user to determine how to proceed. 

I tried to do a little better here;
we know that the content read from the DDB table does not meet the library's expectations.

I am trying to suggest to the user possible causes for this expectation failure;
namely, something other than the MPL wrote to the Index or Mutation Commitment,
and while doing so, 
corrupted the item.

Reply:
How did we validated Commitment and Index previously in Describe mutation? 

If the commitment and index was invalid would a different error be thrown in previous release?

Will this (line 87 to line 102) be a breaking change?

Reply:
I actually think it may have been a bug/miss in mutations. 
I don't think anyone has ever hit this, because they never had an invalid item.

But I'll need to review the file.

Reply:
The error in mutations/mutations would be:
```
                          => Types.KeyStoreAdminException(
                              message := "Could not JSON Deserialize: original properties. " + e.ToString()));
```
Except substitute `original` for which ever field is invalid UTF-8.
I think the error here in HV-2 is actually much better,
as it details which item was invalid.

Unfortunately, 
Smithy-Dafny & Dafny obscure stack traces, 
which make it impossible for the customer to see what went wrong.